### ISSUE TITLE
Warning: Encountered known unsupported method torch.randn

### ISSUE THREAD

**Issue by kshamaramesh** [NVIDIA-AI-IOT/torch2trt#186]
Trying to convert the pytorch model to tensorRT using torch to trt encountered  with this problem **Warning: Encountered known unsupported method torch.randn**

Please find the code snippet below

`class VAEGen(nn.Module):
    # VAE architecture
    def __init__(self, input_dim):
        super(VAEGen, self).__init__()
        # dim = params['dim']
        # n_downsample = params['n_downsample']
        # n_res = params['n_res']
        # activ = params['activ']
        # pad_type = params['pad_type']

        # content encoder
        self.enc = ContentEncoder(2, 4, 3, 64, 'in', 'relu', pad_type='reflect')
        self.dec = Decoder(2, 4, self.enc.output_dim, 3, res_norm='in', activ='relu', pad_type='reflect')

    def forward(self, images):
        # This is a reduced VAE implementation where we assume the outputs are multivariate Gaussian distribution with mean = hiddens and std_dev = all ones.
        hiddens,_ = self.encode(images)
        if self.training == False:
            noise = Variable(torch.randn(hiddens.size()).cuda(hiddens.data.get_device()))
            images_recon = self.decode(hiddens + noise)
        else:
            images_recon = self.decode(hiddens)
        return images_recon, hiddens

    def encode(self, images):
        hiddens = self.enc(images)
        noise = Variable(torch.randn(hiddens.size()).cuda(hiddens.data.get_device()))
        return hiddens, noise

    def decode(self, hiddens):
        images = self.dec(hiddens)
        return images`

**Comment by jaybdub** [comment]
Hi kshamaramesh,

Thanks for reaching out!

I would recommend optimizing just the deterministic portions (encoder and decoder).

Alternatively, if you're able to re-organize the model so that the random data is provided as an input to the module, that may work also.

Please let me know if you have any questions.

Best,
John
